src/Sulu/Component/Content/Mapper/ContentMapper.php

```diff
@@ -1892,10 +1951,24 @@ private function deleteRecursively(NodeInterface $node, $dereference = false)
             }
 
             if ($this->nodeHelper->hasSuluNodeType($child, array('sulu:path'))) {
-                $this->deleteRecursively($child);
+                $this->deleteRecursively($child, $webspace);
```

Comment:
Hmm. Well spotted. I guess it should, so:

- If the node is a `sulu:path` then we should recursively delete it
- We should remove any references to this route if `dereference = true`.

Makes sense I think.